autokeras/hypermodel/block.py

```diff
@@ -0,0 +1,599 @@
+                          [16, 32, 64],
+                          default=32),
+                kernel_size,
+                padding=self._get_padding(kernel_size, output_node))(output_node)
+            output_node = conv(
+                hp.Choice('filters_{i}_2'.format(i=i),
+                          [16, 32, 64],
+                          default=32),
+                kernel_size,
+                padding=self._get_padding(kernel_size, output_node))(output_node)
+            output_node = pool(
+                kernel_size - 1,
+                padding=self._get_padding(kernel_size - 1, output_node))(output_node)
+        return output_node
+
+    @staticmethod
+    def _get_padding(kernel_size, output_node):
+        if (kernel_size * 2 <= output_node.shape[1] and
+                kernel_size * 2 <= output_node.shape[2]):
+            return 'valid'
+        return 'same'
+
+
+class ResNetBlock(HyperBlock, resnet.HyperResNet):
+    """HyperBlock for ResNet.
+
+    # Arguments
+        version: Str. 'v1', 'v2' or 'next'. The type of ResNet to use.
+            If left unspecified, it will be tuned automatically.
+        pooling: Str. 'avg', 'max'. The type of pooling layer to use.
+            If left unspecified, it will be tuned automatically.
+    """
+
+    def __init__(self,
+                 version=None,
+                 pooling=None,
+                 **kwargs):
+        super().__init__(include_top=False, input_shape=(10,), **kwargs)
+        self.version = version
+        self.pooling = pooling
+
+    def build(self, hp, inputs=None):
+        self.input_tensor = nest.flatten(inputs)[0]
+        self.input_shape = None
+
+        hp.Choice('version', ['v1', 'v2', 'next'], default='v2')
+        hp.Choice('pooling', ['avg', 'max'], default='avg')
+
+        hp.values['version'] = self.version or hp.values['version']
+        hp.values['pooling'] = self.pooling or hp.values['pooling']
+
+        model = super(ResNetBlock, self).build(hp)
+        return model.outputs
+
+
+class XceptionBlock(HyperBlock, xception.HyperXception):
+    """XceptionBlock.
+
+    An Xception structure, used for specifying your model with specific datasets.
+
+    The original Xception architecture is from https://arxiv.org/abs/1610.02357.
+    The data first goes through the entry flow, then through the middle flow which
+    is repeated eight times, and finally through the exit flow.
+
+    This XceptionBlock returns a similar architecture as Xception except without
+    the last (optional) fully connected layer(s) and logistic regression.
+    The size of this architecture could be decided by `HyperParameters`, to get an
+    architecture with a half, an identical, or a double size of the original one.
+
+    # Arguments
+        activation: Str. 'selu' or 'relu'. If left unspecified, it will be tuned
+            automatically.
+        conv2d_num_filters: Int. If left unspecified, it will be tuned automatically.
```

Comment:
Not describing an argument in detail is ok if the name of the argument is explicit on its own. In this case, though, it is not clear from the name only what this argument does.